### PR TITLE
Add a Nullable wrapper type to handle NULLs.

### PR DESCRIPTION
**Pull request by andy-thomason**
Solution to https://github.com/extendr/extendr/issues/138

Wrapper for handling potentially NULL values.

```
    let s1 = r!(1);
    let n1 = <Nullable<i32>>::from_robj(&s1)?;
    assert_eq!(n1, Nullable::NotNull(1));
    let snull = r!(NULL);
    let nnull = <Nullable<i32>>::from_robj(&snull)?;
    assert_eq!(nnull, Nullable::Null);

    assert_eq!(r!(Nullable::<i32>::Null), r!(NULL));
    assert_eq!(r!(Nullable::<i32>::NotNull(1)), r!(1));
```

### CODE REVIEW

**Comment by clauswilke** [comment]
To be clear how this interacts with `NA`, could you add an example where a nullable value is set to `NA`?

Would it be something like this?

```rust
let s1 = r!(NA);
let n1 = <Nullable<Option<i32>>>::from_robj(&s1)?;
assert_eq!(n1, Nullable::NotNull(None));
```

File: extendr-api/src/wrapper.rs [hunk 1]
```diff
@@ -542,10 +563,11 @@ impl Robj {
...
     // /// use extendr_api::prelude::*;
     // /// test! {
-    // /// let builtin = r!(Primitive("+"));
-    // /// let special = r!(Primitive("if"));
-    // /// assert_eq!(builtin.sexptype(), libR_sys::BUILTINSXP);
-    // /// assert_eq!(special.sexptype(), libR_sys::SPECIALSXP);
+    // ///  let builtin = r!(Primitive("+"));
+    // ///  let special = r!(Primitive("if"));
+    // ///  assert_eq!(builtin.sexptype(), libR_sys::BUILTINSXP);
+    // ///  assert_eq!(special.sexptype(), libR_sys::SPECIALSXP);
```
Comment (clauswilke) [inline extendr-api/src/wrapper.rs:569]: Should this be indented four spaces?

**Review by clauswilke — APPROVED** [review]

**Review by Ilia-Kosenkov — APPROVED** [review]
No objections from me, but I agree with @clauswilke that at least one test showing how it works (and verifying conversions) should be added.

**Comment by andy-thomason** [comment]
I'll try to get an NA example in.as in r!(NA_INTEGER).

**Pull request merged**